src/runtime/hexagon/hexagon_buffer_manager.h

```diff
@@ -40,14 +40,19 @@ class HexagonBufferManager {
    * \param ptr Address of the HexagonBuffer as returned by `AllocateHexagonBuffer`.
    */
   void FreeHexagonBuffer(void* ptr) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    // Check the list of buffers that were still allocated at the time of release.  If this buffer
+    // is in that list, this is a no-op as it is being freed as part of another object teardown.
+    auto it_released = std::find(released_buffers_.begin(), released_buffers_.end(), ptr);
+    if (it_released != released_buffers_.end()) {
+      released_buffers_.erase(it_released);
+      return;
+    }
```

Comment:
I think what we want here is to not allow any new allocations when released_buffers_ has any values.

In the current checked in code, that's what would happen because there would be no runtime_hexbuffs

Reply:
There is no longer any `released_buffers_` structure so the original feedback is no longer valid.

Reply:
> I think what we want here is to not allow any new allocations when released_buffers_ has any values.

This PR will fail / assert if we `Enable` the HexagonBufferManager with unfreed buffers.